File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/TeamProfileChangeEvent.java
package work.socialhub.kslack.entity.event;

import java.util.List;

/**
 * The team_profile_change event is sent to all connections for a workspace
 * when Workspace Admin updates the field definitions in the profile.
 * Only the modified field definitions are included in the payload.
 * Where appropriate, clients should update to reflect new changes immediately.
 * <p>
 * https://api.slack.com/events/team_profile_change
 */
public class TeamProfileChangeEvent implements Event {

    public static final String TYPE_NAME = "team_profile_change";

    private final String type = TYPE_NAME;
    private Profile profile;

    public String getType() {
        return this.type;
    }

    public Profile getProfile() {
        return this.profile;
    }

    public void setProfile(Profile profile) {
        this.profile = profile;
    }

    public static class Profile {
        private List<Field> fields;

        public List<Field> getFields() {
            return this.fields;
        }

        public void setFields(List<Field> fields) {
            this.fields = fields;
        }
    }

    public static class Field {
        private String id;

        public String getId() {
            return this.id;
        }

        public void setId(String id) {
            this.id = id;
        }
        // TODO: other attributes
    }

}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/TeamProfileChangeEvent.kt
package work.socialhub.kslack.entity.event

/**
 * The team_profile_change event is sent to all connections for a workspace
 * when Workspace Admin updates the field definitions in the profile.
 * Only the modified field definitions are included in the payload.
 * Where appropriate, clients should update to reflect new changes immediately.
 *
 *
 * https://api.slack.com/events/team_profile_change
 */
class TeamProfileChangeEvent : Event {
    override val type: String = TYPE_NAME
    var profile: Profile? = null

    class Profile {
        var fields: Array<Field>? = null
    }

    class Field {
        // TODO: other attributes
        var id: String? = null
    }

    companion object {
        const val TYPE_NAME: String = "team_profile_change"
    }
}